lib/src/container/store.rs

```diff
@@ -983,6 +983,40 @@ pub async fn copy(
     Ok(())
 }
 
+/// Iterate over deployment commits, returning the manifests from
+/// commits which point to a container image.
+fn list_container_deployment_manifests(
```

Comment:
Code LGTM. However I have a general observation and alternative suggestion on GC logic.

I have a feeling that we should try to rework all the code behind `gc_image_layers()` so that it becomes infallible (or at least provide an infallible version to consumers). The idea is that a minor unexpected state or a single error in the GC logic can quickly spiral down into a instance with a full sysroot and hard to recover.
To that extent, it is probably more useful to keep going cleaning up whatever we can without failing, and log error messages if we encounter any failures.

Reply:
Hmm.  I see your point.  Arguably we should do the same inside `ostree_repo_prune()` for the objects right?

I think the most cases of failures here are going to be filesystem corruption...in which case, trying to continue probably won't help much.

But in the case of a bug in the logic where we too-eagerly pruned a layer in earlier code; yeah I'd agree continuing makes sense because the user can always re-pull that layer.  This really gets into the need for a similar `ostree fsck --repair` type logic.

Reply:
OK one thing I did verify is that the key bit of `repo.set_ref_immediate(None, layer_ref.as_str(), None, cancellable)?;` [is already idempotent](https://github.com/ostreedev/ostree/blob/8cb50c16a05cbd7046077036df4996b2000552f4/src/libostree/ostree-repo-refs.c#L1164).  So I think that addresses a major source of possible logic errors already.

This needs some thought; filed https://github.com/ostreedev/ostree-rs-ext/issues/407

Reply:
Yes I guess the main part of this would be on objects pruning instead, I agree.

Overall on GC there are many little things that could be just barely misaligned enough to cascade into larger issues. So it seems a good idea to at least trying to keep going whenever possible and clean at least some of the things.